hbase-common/src/main/java/org/apache/hadoop/hbase/HBaseConfiguration.java

```diff
@@ -85,16 +85,69 @@ public static Configuration addHbaseResources(Configuration conf) {
     return conf;
   }
 
+  public static Configuration create() {
+    return create(true);
+  }
+
   /**
    * Creates a Configuration with HBase resources
    * @return a Configuration with HBase resources
    */
-  public static Configuration create() {
+  public static Configuration create(boolean lowLimits) {
     Configuration conf = new Configuration();
     // In case HBaseConfiguration is loaded from a different classloader than
     // Configuration, conf needs to be set with appropriate class loader to resolve
     // HBase resources.
     conf.setClassLoader(HBaseConfiguration.class.getClassLoader());
+
+
+    if (lowLimits) {
+
+      // can't set this for every test currently
+      // conf.set("hbase.regionserver.hostname", "127.0.0.1");
+
+      conf.setInt("hbase.hfilearchiver.thread.pool.max", 2);
+      conf.setInt("hbase.loadincremental.threads.max", 3);
+
+
+      conf.setInt("hbase.client.sync.wait.timeout.msec", 30000);
+      conf.setInt("zookeeper.recovery.retry", 5);
+      conf.setInt("hbase.client.retries.number", 10);
+      conf.setInt("hbase.hstore.open.and.close.threads.max", 1);
+      conf.setInt(HConstants.REGION_SERVER_REPLICATION_HANDLER_COUNT, 2);
+
+      conf.setInt(HConstants.HBASE_CLIENT_MAX_TOTAL_TASKS, 3);
+
+      conf.setInt(HConstants.REGION_SERVER_HIGH_PRIORITY_HANDLER_COUNT, 2);
```

Comment:
Probably keep these high because in certain circumstance test can lock up because we are starved for priority handlers.